lib/matplotlib/ticker.py

```diff
@@ -1574,7 +1574,7 @@ def tick_values(self, vmin, vmax):
         """
         if self.nbins is None:
             return self.locs
-        step = max(int(0.99 + len(self.locs) / float(self.nbins)), 1)
+        step = max(int(np.ceil(len(self.locs) / self.nbins)), 1)
```

Comment:
Won't this change the results in rare cases?  It does look nicer, though.

Reply:
1) I think the code really, really looks like it was written by someone who meant ceil but didn't remember that function existed.
2) For the change to actually matter you need a) to use bins with FixedLocator, which seems to be quite rare in my experience, and b) at least 99 ticks and 100 bins (to fall in the "edge" region), which is also quite rare...

Reply:
I'll go along with this one.